packages/tpetra/core/src/Tpetra_Details_FixedHashTable_decl.hpp

```diff
@@ -328,7 +328,11 @@ class FixedHashTable {
     // numPairs_ field (remembering the size of the input array of
     // keys).
     if (this->hasContiguousValues ()) {
-      return val_.extent (0) + static_cast<offset_type> (lastContigKey_ - firstContigKey_);
+       // NOTE: We need this check for a newly initialied FHT.
+         if(minVal_ == ::Kokkos::ArithTraits<ValueType>::max())
+             return val_.extent (0);
+          else
+             return val_.extent (0) + static_cast<offset_type> (lastContigKey_ - firstContigKey_);
     }
```

Comment:
I would keep the code as before, but set both `lastContigKey_` and `firstContigKey_` to the same value by default. This would keep the logic simple (and fast).

For now, `firstContigKey_` is set to `max` and `lastContigKey_` set to `min`, but I am not sure to understand the usefulness of such an initial setting...

```suggestion
             return val_.extent (0) + static_cast<offset_type> (lastContigKey_ - firstContigKey_);
    }
```